node/src/neighborhood/neighborhood_database.rs

```diff
@@ -532,44 +532,43 @@ mod tests {
     #[test]
     fn node_by_ip_mut_works() {
         let this_node = make_node_record(1234, true);
+        let this_node_ip_addr = this_node.node_addr_opt().unwrap().ip_addr();
         let this_node_original_version = this_node.inner.version;
         let one_node = make_node_record(4567, true);
+        let one_node_ip_addr = one_node.node_addr_opt().unwrap().ip_addr();
         let one_node_original_version = one_node.inner.version;
         let another_node = make_node_record(5678, true);
+        let another_node_ip_addr = another_node.node_addr_opt().unwrap().ip_addr();
         let mut subject = db_from_node(&this_node);
-
         subject.add_node(one_node.clone()).unwrap();
+        fn mutable_node(db: &mut NeighborhoodDatabase, ip_addr: IpAddr) -> &mut NodeRecord {
+            db.node_by_ip_mut(ip_addr).unwrap()
+        }
+        fn node_version(db: &NeighborhoodDatabase, ip_addr: IpAddr) -> u32 {
+            db.node_by_ip(ip_addr).as_ref().unwrap().inner.version
+        }
```

Comment:
I developed a certain habit and I don't mean to impose it onto you but I like when any inner functions stand at the top of the outer function, as the first lines, and after that I leave a blank line there to end that area after which I can start writing the actual code of the outer function (or of the test) .

Thinking more about it, I think here this is a special situation where it makes some sense if them stay as central as possible because they represent something like an Act.

Reply:
Yeah, I kind of like these functions to be defined right here near where they're used.